pytapo/__init__.py

```diff
@@ -71,73 +73,71 @@ def responseIsOK(self, res):
                 + str(res.status_code)
             )
         try:
-            data = json.loads(res.text)
+            data = res.json()
             return res and data and data["error_code"] == 0
         except Exception as e:
             raise Exception("Unexpected response from Tapo Camera: " + str(e))
 
-    def getOsd(self, raiseException=False):
+    def performRequest(self, inData, loginRetryCountdown=5):
         self.ensureAuthenticated()
         url = self.getHostURL()
-        data = {
-            "method": "get",
-            "OSD": {"name": ["date", "week", "font"], "table": ["label_info"]},
-        }
         res = requests.post(
-            url, data=json.dumps(data), headers=self.headers, verify=False
+            url, data=json.dumps(inData), headers=self.headers, verify=False
         )
-        data = json.loads(res.text)
-        if self.responseIsOK(res):
-            return json.loads(res.text)
-        else:
-            if raiseException:
+        data = res.json()
+        if not self.responseIsOK(res):
```

Comment:
I am not sure how res.json behaves if the response is not a json. Could we move `data = res.json()` after if not self.responseIsOK(res): instead?

Reply:
I see